stm/asyncTaskQueue.mli

```diff
@@ -6,79 +6,129 @@
 (*         *       GNU Lesser General Public License Version 2.1        *)
 (************************************************************************)
 
+(** A status is
+    - `Fresh when XXX
+    - `Old x when XXX
+  *)
 type 'a worker_status = [ `Fresh | `Old of 'a ]
 
+(** The [Task] module type defines an abstract message-processing
+    entity. *)
 module type Task = sig
 
+  (** Identifier of a task, for workers that can process several
+      jobs. Usually an integer or any other unique identifier. *)
   type task
+
+  (** XXX No idea what this does *)
   type competence
 
-  (* Marshallable *)
+  (** Type of input and output data for workers.
+
+      The data must be marshallable as it send through the network
+      using [Mashal] . *)
   type request
   type response
 
-  val name : string ref (* UID of the task kind, for -toploop *)
+  (** UID of the task kind, for -toploop *)
+  val name : string ref
+  (** Extra arguments of the task kind, for -toploop *)
   val extra_env : unit -> string array
 
-  (* run by the master, on a thread *)
+  (** {5} Client API, it is run by the master, on a thread *)
+
+  (** [request_of_task status tid] checks if the worker has already
+      processed [tid] *)
   val request_of_task : competence worker_status -> task -> request option
+
+  (** [task_match status tid] XXX *)
   val task_match : competence worker_status -> task -> bool
+
+  (** [use_response status tid out] notifies the worker that [tid] can
+      be freeded? *)
   val use_response :
     competence worker_status -> task -> response ->
       [ `Stay of competence * task list | `End ]
+
+  (** [on_marshal_error err_msg tid] notifies of marshaling failure. *)
   val on_marshal_error : string -> task -> unit
+
+  (** [on_task_cancellation_or_expiration_or_slave_death tid] XXX  *)
```

Comment:
These functions are meant to parametrize the worker manager on the actions to be taken when things go wrong or are cancelled (you can kill a worker in CoqIDE, or using kill -9...)
E.g. master can decide to inhabit the (delegate) Future.t with a closure (to be run in master), i.e. make the document still checkable.  This is what I do for marshaling errors.